pkg/controller/operators/catalog/operator.go

```diff
@@ -622,6 +628,35 @@ func (o *Operator) syncConnection(logger *logrus.Entry, in *v1alpha1.CatalogSour
 	return
 }
 
+func (o *Operator) waitForBackingPod(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *v1alpha1.CatalogSource, continueSync bool, syncError error) {
+	out = in.DeepCopy()
+
+	selector := map[string]string{reconciler.CatalogSourceLabelKey: in.GetName()}
+	pods, err := o.lister.CoreV1().PodLister().Pods(in.GetNamespace()).List(labels.SelectorFromSet(selector))
+	if err != nil {
+		continueSync = false
+		syncError = fmt.Errorf("error while examining catalog source pod: %v", err)
+		return
+	}
+
+	if len(pods) > 0 {
+		for _, cond := range pods[0].Status.Conditions {
```

Comment:
Here we are getting an array of pods but only checking the condition of the pod with index 0 so the first one. Can we can always expect to get one pod in the `pods` list, or if the first one is ready then the others must be as well? If we only look at the one, can we use a `Get` instead of a `List` when making the request on L635?

Reply:
Hrm, good point. I'm pretty sure the mapping is always 1:1...

Reply:
Turns out that Get can only be queried by name, so I'm sticking with List here.